### PR TITLE
Revert "Add ability to use mchirp, eta as variable/frozen params"

### PR DESCRIPTION
**Pull request by cmbiwer**
Reverts ligo-cbc/pycbc#1069

### CODE REVIEW

File: pycbc/waveform/generator.py [hunk 1]
```diff
@@ -99,95 +98,21 @@ def generate(self, *args):
...
-
-    Derived parameters not understood by `get_fd_waveform` may be used as
-    variable args and/or frozen parameters, as long as they can be converted
-    into parameters that `get_fd_waveform` can use. For example, `mchirp` and
-    `eta` (currently, the only supported derived parameters) may be used as
-    variable/frozen params; these are converted to `mass1` and `mass2` prior to
-    calling the waveform generator function.
+class FDomainCBCGenerator(BaseGenerator):
+    """Uses waveform.get_fd_waveform as a generator function to create frequency-
+    domain CBC waveforms in the radiation frame; i.e., with no detector response
```
Comment (ahnitz) [inline pycbc/waveform/generator.py]: __[flake8]__
* [E501] line too long (80 > 79 characters)

File: pycbc/waveform/generator.py [hunk 2]
```diff
@@ -198,58 +123,25 @@ class FDomainCBCGenerator(BaseCBCGenerator):
...
-     'mass2': 1.7230475324955525,
-     'mchirp': 1.5}
     """
     def __init__(self, variable_args=(), **frozen_params):
         super(FDomainCBCGenerator, self).__init__(waveform.get_fd_waveform,
             variable_args=variable_args, **frozen_params)
 
-
-class TDomainCBCGenerator(BaseCBCGenerator):
+class TDomainCBCGenerator(BaseGenerator):
```
Comment (ahnitz) [inline pycbc/waveform/generator.py]: __[flake8]__
* [E302] expected 2 blank lines, found 1

File: pycbc/waveform/generator.py [hunk 3]
```diff
@@ -198,58 +123,25 @@ class FDomainCBCGenerator(BaseCBCGenerator):
...
-    domain CBC waveforms in the radiation frame; i.e., with no detector
-    response function applied. For more details, see `BaseGenerator`.
-
-    Derived parameters not understood by `get_td_waveform` may be used as
-    variable args and/or frozen parameters, as long as they can be converted
-    into parameters that `get_td_waveform` can use. For example, `mchirp` and
-    `eta` (currently, the only supported derived parameters) may be used as
-    variable/frozen params; these are converted to `mass1` and `mass2` prior to
-    calling the waveform generator function.
+    domain CBC waveforms in the radiation frame; i.e., with no detector response
```
Comment (ahnitz) [inline pycbc/waveform/generator.py]: __[flake8]__
* [E501] line too long (80 > 79 characters)

File: pycbc/waveform/generator.py [hunk 4]
```diff
@@ -198,58 +123,25 @@ class FDomainCBCGenerator(BaseCBCGenerator):
...
+    function applied. For more details, see BaseGenerator.
 
     Examples
     --------
     Initialize a generator:
     >>> generator = waveform.TDomainCBCGenerator(variable_args=['mass1', 'mass2'], delta_t=1./4096, f_lower=30., approximant='TaylorT4')
 
-    Create a waveform with the variable arguments (in this case, mass1, mass2):
+    Create a waveform with the variable arguments (in this casee, mass1, mass2):
```
Comment (ahnitz) [inline pycbc/waveform/generator.py]: __[flake8]__
* [E501] line too long (80 > 79 characters)

**Comment by cmbiwer** [comment]
This PR doesn't actually work, it breaks if you try to provide ``mchirp`` and ``eta`` with pickling errors for instance methods.

If I have time I'll fix it.

**Comment by cmbiwer** [comment]
I found one possible solution to the problem, the functions ``self.mchirp_eta_to_mass1_mass2`` needs to be top-level in ``generator.py``. This is just seems a bit wonky though.

I'll close this PR and include this solution in my next PR.

**Pull request closed without merging**